From b344190aa8dbeca560d26d42c418a3b2fcca82c0 Mon Sep 17 00:00:00 2001
From: rselvaraj2 <rselvaraj2@maxlinear.com>
Date: Wed, 20 Nov 2024 16:50:56 +0800
Subject: [PATCH] WLANRTSYS-85267 Probe responses are malformed in the
 reporting link during critical update got 7.2.60 branch

Issue:
	Probe responses are malformed in the reporting link during critical update

Root Cause:
	CSA, CSW IE construction is not handled properly.

Fix:
	Temporary WAR to prevent malformed probe responses during critical updates by skipping per-sta profile addition in ML IE.

FIXUP to WLANRTSYS-63318 ML CSA with puncturing and BSS critical update
---
 src/ap/mxl_beacon.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/src/ap/mxl_beacon.c b/src/ap/mxl_beacon.c
index 920c71e4b..298085f68 100644
--- a/src/ap/mxl_beacon.c
+++ b/src/ap/mxl_beacon.c
@@ -68,6 +68,11 @@ u8 * mxl_hostapd_eid_ml_probe(struct hostapd_data *hapd, const u8 *probe_req_mle
 	struct hostapd_data *non_tx_probe_req_vap = NULL;
 	int non_tx_bss_idx;
 
+	//Temp-WAR to prevent malformed probe response by skipping Per-STA profile addition during critical updates in ML IE.
+	hapd->mxl_data.ml_ie_csa_ie_len_add = false;
+	hapd->mxl_data.ml_ie_max_chan_sw_time_ie_add = false;
+	hapd->mxl_data.ml_ie_max_chan_sw_time_ie_len_add = false;
+
 	if ((hapd->iconf->multibss_enable) && (probe_req_mle)) {
 		non_tx_bss_idx = mxl_hostap_ml_get_non_tx_bss_idx_from_mlid(hapd,
 									    probe_req_mle,
-- 
2.43.0

